components/scripts/gradle/gradle-init-scripts/configure-gradle-enterprise.gradle

```diff
@@ -55,15 +55,18 @@ void configureGradleEnterprise(gradleEnterprise) {
                 server = projectProperties.get("com.gradle.enterprise.build_validation.server")
             }
 
-            if (!server) {
+            if (!server && !gradle.startParameter.noBuildScan) {
                 throw new IllegalStateException("A Gradle Enterprise server URL has not been configured.", null)
             }
 
             // captureTaskInputFiles = true (too late to be set here for Gradle 5, set via sys prop)
             uploadInBackground = false
             publishAlways()
         }
-        addCustomData(buildScan)
+
+        if (!gradle.startParameter.noBuildScan) {
```

Comment:
This needed to be added because `addCustomData(...)` tries to add links to the Gradle Enterprise dashboard and was causing a NPE.

Reply:
No longer relevant as I have changed the code.